Pidaq/UI/Stability/stability_dash/callbacks/control_callbacks.py

```diff
@@ -0,0 +1,138 @@
+import json
+import os
+
+import dash
+from dash.dependencies import Input, Output, State
+from dash.dash import no_update 
+
+from stability_dash.app import app
+from stability_dash.layout.control import control_layout
+
+
+@app.callback(
+    Output("save-load-connection", "value"),
+    [Input("save-profile", "n_clicks")],
+    [State("profile-name", "value"),
+     State("runtime", "value"),
+     State("timestep", "value"),
+     State("displacement-step", "value"),
+     State("velocity-step", "value"),
+     State("acceleration-step", "value")]
+)
+def save_profile(save_clicks, profile_name, runtime, timestep, displacement, velocity, acceleration):
+    
+    if not save_clicks:
+        raise dash.exceptions.PreventUpdate
+
+    folder_path = os.getcwd()
+    path_to_file = os.path.join(folder_path, "profiles.json")
+
+    if not os.path.isfile(path_to_file):
+        profiles_json = {}
+        profiles_json['profiles'] = []
+        profiles_json['profiles'].append({
+        'name': profile_name,
+        'runtime': runtime,
+        'timestep': timestep,
+        'displacement-step' : displacement,
+        'velocity-step' : velocity,
+        'acceleration-step' : acceleration
+        })
+
+        with open(path_to_file, 'w') as profiles_file:
+            json.dump(profiles_json, profiles_file, indent=4)
+
+    else:
+        with open(path_to_file, 'r') as profiles_file:
+            profiles_json = json.load(profiles_file)
+            for profile in profiles_json['profiles']:
+                if profile['name'] == profile_name:
+                    profile['runtime'] = runtime
+                    profile['timestep'] = timestep
+                    profile['displacement-step'] = displacement
+                    profile['velocity-step'] = velocity
+                    profile['acceleration-step'] = acceleration
+                    with open(path_to_file, 'w') as profiles_file:
+                        json.dump(profiles_json, profiles_file, indent=4)
+                    break
```

Comment:
This is good and works - although it may have been easier to key the profile by the name, so you have a constant time lookup instead of iterating through every profile. Then, you can also avoid having the if logic - it would simply be something like:

The json file:
```
{ "profiles": 
    {
    "Test-1": {

         },
    "Test-2":

    }

```

Writing to profile:
```
profiles_json['profiles']['Test-1'] = {
"runtime" : something,
"timestep" : something_else, 
etc
}
```
Since you are always overwriting the profile anyways, there is no need to check if it exists or not. 

P.S. I hate trying to type json in markdown - why cant I tab bro

@Burke-Daniel @afhassan Thoughts?

Reply:
I believe lines 45-69 can be collapsed from 24 lines to 12-15